### PR TITLE
[ReportBundle] Remove bi-directional dependency on the CoreBundle

### PR DESCRIPTION
**Pull request by adamelso**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | 
| License       | MIT
| Doc PR        | n/a

Thanks for merging the [other pull request](https://github.com/Sylius/Sylius/pull/3212), now the real issue - The ReportBundle is completely unusable because of the following error:

 > Fatal error: Uncaught exception 'Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException' with message 'The service "sylius.form.type.report" has a dependency on a non-existent service "sylius.registry.report.data_fetcher"

These missing services are registered by the CoreBundle, so I've moved them to the ReportBundle.

Additionally, I've grouped and sorted the service definitions in alphabetical order (it was getting really confusing otherwise).

Don't merge yet, though, I need to do some more work. Cheers.

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
I added the [WIP] tag to this PR. Thanks Adam! Let us know when it is ready for review.

**Comment by adamelso** [comment]
Thanks Pawel, this is ready now. I managed to register the bundle in my project okay.

**Comment by pjedrzejewski** [comment]
Nice! Requires a rebase though. :)

**Comment by adamelso** [comment]
@pjedrzejewski Done :)  Though, I see that #3110 is merged, will I need to do it again? Incidently, the build is failing on PHP 5.3 as Doctrine Migrations now requires 5.4.

**Comment by pamil** [comment]
@adamelso Yes, I'm afraid it will need one another rebase :)

File: src/Sylius/Bundle/ReportBundle/composer.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
         "stof/doctrine-extensions-bundle": "~1.1",
         "sylius/resource-bundle":          "0.16.*",
         "sylius/money-bundle":             "0.16.*",
-        "sylius/report":                   "0.16.*"
+        "sylius/registry":                 "0.16.*",
+	"sylius/report":                   "0.16.*"
```
Comment (adamelso) [inline src/Sylius/Bundle/ReportBundle/composer.json]: @pamil okay great, because I need to correct this line here :D

**Comment by adamelso** [comment]
alright, good to go! :)

**Comment by stloyd** [comment]
:+1:

**Comment by pjedrzejewski** [comment]
Thank you Adam! :+1:

**Pull request merged**